### ISSUE TITLE
IP address never updated on freedns.afraid.org

### ISSUE THREAD

**Issue by jcourcel** [troglobit/inadyn#97]
Hi

When my Internet Provider change my IP address, it is never update on the FreeDNS.afraid.org.  I always have the message "Update URL query failed" from the inadyn.  When I log to the site freedns.afraid.org and go under the "Dynamic DNS", if I click to the button "Direct URL", my IP address is updated.  Do I mist something in my config file?  Do you have any idea what is the problem?

I am running inadyn version 1.99.9 under Debian 7.

This is my config file:
username MyEmailAddress
password MyPassword
system default@freedns.afraid.org
alias mahingan.homenet.org
ssl
log_file /var/log/inadyn.log
period 600
background
verbose 5

This is the log I got:
Mon Jun  8 20:12:57 2015: Inadyn version 1.99.9 -- Dynamic DNS update client.
Mon Jun  8 20:12:57 2015: Resolving hostname mahingan.homenet.org => IP# 206.47.22.34
Mon Jun  8 20:12:57 2015: Checking for IP# change, connecting to checkip.dyndns.org(216.146.38.70:80)
Mon Jun  8 20:12:57 2015: Querying DDNS checkip server for my public IP#: GET / HTTP/1.0
Host: checkip.dyndns.org
User-Agent: inadyn/1.99.9 troglobit@gmail.com

Mon Jun  8 20:12:57 2015: Checked my IP, return code: 0
Mon Jun  8 20:12:57 2015: IP server response:
Mon Jun  8 20:12:57 2015: HTTP/1.1 200 OK
Content-Type: text/html
Server: DynDNS-CheckIP/1.0
Connection: close
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 104

<html><head><title>Current IP Check</title></head><body>Current IP Address: 199.102.96.9</body></html>
Mon Jun  8 20:12:57 2015: Current public IP# 199.102.96.9
Mon Jun  8 20:12:57 2015: Update forced for alias mahingan.homenet.org, new IP# 199.102.96.9
Mon Jun  8 20:12:57 2015: Sending IP# update to DDNS server, connecting to freedns.afraid.org(69.197.18.161:443)
Mon Jun  8 20:12:57 2015: Sending IP# update to DDNS server, initiating HTTPS ...
Mon Jun  8 20:12:58 2015: SSL connection using DHE-RSA-AES256-SHA
Mon Jun  8 20:12:58 2015: Server certificate -- subject: /OU=Domain Control Validated/OU=EssentialSSL/CN=freedns.afraid.org; issuer: /C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=EssentialSSL CA
Mon Jun  8 20:12:58 2015: Sending update URL query, connecting to freedns.afraid.org(69.197.18.161:443)
Mon Jun  8 20:12:58 2015: Sending update URL query, initiating HTTPS ...
Mon Jun  8 20:12:58 2015: SSL connection using DHE-RSA-AES256-SHA
Mon Jun  8 20:12:58 2015: Server certificate -- subject: /OU=Domain Control Validated/OU=EssentialSSL/CN=freedns.afraid.org; issuer: /C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=EssentialSSL CA
Mon Jun  8 20:12:58 2015: Successfully sent DDNS update using HTTPS!
Mon Jun  8 20:12:59 2015: Successfully received DDNS update response (184 bytes) using HTTPS!
Mon Jun  8 20:12:59 2015: Update URL query failed
Mon Jun  8 20:12:59 2015: Sending alias table update to DDNS server: 
Mon Jun  8 20:12:59 2015: DDNS server response: 

Thanks a lot for your help.
Best Regards
Jocelyn

**Comment by troglobit** [comment]
Hi Jocelyn!

Sorry to hear of your problems.  FreeDNS is one of the DDNS providers that I myself actually have an account on and can test, but I've never had any problems like you describe.  If I build the latest Inadyn from GitHub I get the following result for my own domain:

    Tue Jun  9 07:54:15 2015: Sending update URL query, connecting to freedns.afraid.org(204.140.20.21:443)
    Tue Jun  9 07:54:15 2015: Sending update URL query, initiating HTTPS ...
    Tue Jun  9 07:54:16 2015: SSL connection using RSA_AES_128_CBC_SHA1
    Tue Jun  9 07:54:16 2015: SSL server cert subject: C=, ST=, L=, O=, OU=Domain Control Validated, CN=freedns.afraid.org/Email=
    Tue Jun  9 07:54:16 2015: SSL server cert issuer: C=GB, ST=Greater Manchester, L=Salford, O=COMODO CA Limited, OU=, CN=EssentialSSL CA/Email=
    Tue Jun  9 07:54:16 2015: Successfully sent DDNS update using HTTPS!
    Tue Jun  9 07:54:16 2015: Successfully received DDNS update response (281 bytes) using HTTPS!
    Tue Jun  9 07:54:16 2015: Successfully sent DDNS update using HTTPS!
    Tue Jun  9 07:54:17 2015: Successfully received DDNS update response (201 bytes) using HTTPS!
    Tue Jun  9 07:54:17 2015: Successful alias table update for troglobit.homenet.org => new IP# 109.228.161.89

That last line indicates successful transfer of the IP address to the server.

My inadyn.conf looks like this, with my credentials X:ed out

    period         600
    forced-update 604800
    cache-dir /etc/inadyn

    system default@freedns.afraid.org
      ssl
      username XXXXXXXX
      password XXXXXXXX
      alias troglobit.homenet.org

Are you sure your credentials are OK?

**Comment by jcourcel** [comment]
Hi Joachim,

Thank a lot for your quick answer!  I appreciate.

You are right, I verify my credentials and found I put my email address instead of my freedns UserID.  I restart inadyn, and this is the log I got.  I think it will works fine when my Internet Provider will change my IP address.  I feel so stupid to have disturb you about this issue.  Again, a big thanks for your help and for sharing your good application!

Have a nice day.
Best Regards
Jocelyn

Tue Jun  9 09:06:17 2015: Inadyn version 1.99.9 -- Dynamic DNS update client.
Tue Jun  9 09:06:17 2015: Resolving hostname mahingan.homenet.org => IP# 199.102.96.9
Tue Jun  9 09:06:17 2015: Checking for IP# change, connecting to checkip.dyndns.org(216.146.38.70:80)
Tue Jun  9 09:06:17 2015: Querying DDNS checkip server for my public IP#: GET / HTTP/1.0
Host: checkip.dyndns.org
User-Agent: inadyn/1.99.9 troglobit@gmail.com

Tue Jun  9 09:06:17 2015: Checked my IP, return code: 0
Tue Jun  9 09:06:17 2015: IP server response:
Tue Jun  9 09:06:17 2015: HTTP/1.1 200 OK
Content-Type: text/html
Server: DynDNS-CheckIP/1.0
Connection: close
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 104

<html><head><title>Current IP Check</title></head><body>Current IP Address: 199.102.96.9</body></html>
Tue Jun  9 09:06:17 2015: No IP# change detected, still at 199.102.96.9
Tue Jun  9 09:06:17 2015: Update forced for alias mahingan.homenet.org, new IP# 199.102.96.9
Tue Jun  9 09:06:17 2015: Sending IP# update to DDNS server, connecting to freedns.afraid.org(69.197.18.161:443)
Tue Jun  9 09:06:18 2015: Sending IP# update to DDNS server, initiating HTTPS ...
Tue Jun  9 09:06:19 2015: SSL connection using DHE-RSA-AES256-SHA
Tue Jun  9 09:06:19 2015: Server certificate -- subject: /OU=Domain Control Validated/OU=EssentialSSL/CN=freedns.afraid.org; issuer: /C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=EssentialSSL CA
Tue Jun  9 09:06:19 2015: Sending update URL query, connecting to freedns.afraid.org(69.197.18.161:443)
Tue Jun  9 09:06:19 2015: Sending update URL query, initiating HTTPS ...
Tue Jun  9 09:06:19 2015: SSL connection using DHE-RSA-AES256-SHA
Tue Jun  9 09:06:19 2015: Server certificate -- subject: /OU=Domain Control Validated/OU=EssentialSSL/CN=freedns.afraid.org; issuer: /C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=EssentialSSL CA
Tue Jun  9 09:06:19 2015: Successfully sent DDNS update using HTTPS!
Tue Jun  9 09:06:19 2015: Successfully received DDNS update response (392 bytes) using HTTPS!
Tue Jun  9 09:06:19 2015: Sending alias table update to DDNS server: GET /dynamic/update.php?=&address=199.102.96.9 HTTP/1.0
Host: freedns.afraid.org
User-Agent: inadyn/1.99.9 troglobit@gmail.com

Tue Jun  9 09:06:19 2015: Successfully sent DDNS update using HTTPS!
Tue Jun  9 09:06:20 2015: Successfully received DDNS update response (199 bytes) using HTTPS!
Tue Jun  9 09:06:20 2015: DDNS server response: HTTP/1.1 200 OK
Server: nginx
Date: Tue, 09 Jun 2015 13:06:20 GMT
Content-Type: text/plain; charset=utf-8
Connection: close
Vary: Accept-Encoding

ERROR: Address 199.102.96.9 has not changed.
Tue Jun  9 09:06:20 2015: Successful alias table update for mahingan.homenet.org => new IP# 199.102.96.9
